### PR TITLE
Clarify self hosted and private urls for providers and add gitlab integration

### PR DESCRIPTION
**Pull request by Letiste**
### What does this PR do?

- Clarify self hosted and private urls in source-code-integration doc
- Add support for GitLab SaaS

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/integrations/guide/source-code-integration.md [hunk 1]
```diff
@@ -181,15 +181,29 @@ COPY .git ./.git
...
 [1]: https://docs.datadoghq.com/integrations/github/
 [2]: https://app.datadoghq.com/integrations/github/
 {{% /tab %}}
+{{% tab "GitLab" %}}
+
+<div class="alert alert-warning">
+Self-hosted instances or private URLs are not supported.
+</div>
+
+If you are a GitLab SaaS user, no setup is required to link your telemetry with your source code.
```
Comment (rpelliard) [inline content/en/integrations/guide/source-code-integration.md:197]: Customers will have code snippets out of the box, but they still need to configure `datadog-ci git-metadata upload` to link telemetry. Should we remove this section, and only have special treatment for GitLab on the `Inline source code` sectio below ?

File: content/en/integrations/guide/source-code-integration.md [hunk 2]
```diff
@@ -245,7 +259,7 @@ You can directly access a trace in its source repository on GitHub in the [Conti
 
 ### Inline source code
 
-If you are a GitHub SaaS user, install Datadog's [GitHub integration][2] to directly inline code snippets from your GitHub repository in your stack traces in [Error Tracking][8]. When specifying permissions on the integration tile, enable Datadog read permissions to **Contents**.
+If you are a GitHub or GitLab SaaS user, configure your [provider integration](#configure-your-repositories) to directly inline code snippets from your repository in your stack traces in [Error Tracking][8].
```
Comment (rpelliard) [inline content/en/integrations/guide/source-code-integration.md:262]: I find it a bit confusing:
- in "GitHub or GitLab SaaS user", it's not clear whether "SaaS" refers to GitLab only or GItHub as well
- I'm not sure "GitLab SaaS" is very commonly used, maybe "gitlab.com" is more explicit ?
- for GitLab, there is no provider integration to configure and it's coming out of the box
GitHub or GitLab SaaS userHow about updating this section to have two separate tabs for GitHub and GitLab ?

**Review by buraizu — APPROVED** [review]
Thanks, LGTM. Let me know if/when this is ready to merge.

**Review by rpelliard — APPROVED** [review]

**Comment by Letiste** [comment]
Hey @buraizu, would you know if it's possible to remove the gap created because of the tabs?
With the GitHub tab, there is a gap before the `1. Navigate ...`
image

With the GitLab tab, there is no gap
image

**Comment by buraizu** [comment]
Hey @Letiste thanks for following up on that. I've been experimenting with this in a local build of the documentation site, but haven't been able to make that space disappear. It seems that this happens in some other places in the documentation as well:
https://docs.datadoghq.com/containers/kubernetes/installation/?tab=operator#container-registries
I'm following up internally with the websites team about this, feel free to ping me in Slack for additional details.

**Comment by buraizu** [comment]
Hey @Letiste, just following up again to confirm that this is expected behavior for a tab with not much content. The only way to remove the blank space would be to add additional content to the tab. As a potential option, I tried updating the second sentence to a note, which seems to help somewhat:
Datadog Source Code Integration 2023-08-10 at 1 30 40 PM
.

Feel free to make any other additions or let me know any feedback.

**Comment by Letiste** [comment]
@buraizu Thank you for looking at this 🙂!
I updated the doc to add the note like you did (I preferred it that way).
It's good on my side so you can merge the PR 😄

**Review by buraizu — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**